runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/feel/FeelEngineWrapper.java

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.jobworker.feel;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.module.scala.DefaultScalaModule$;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.camunda.feel.FeelEngine;
+import org.camunda.feel.impl.SpiServiceLoader;
+import scala.jdk.javaapi.CollectionConverters;
+
+public class FeelEngineWrapper {
+
+  static final String RESPONSE_MAP_KEY = "response";
+  static final String ERROR_VARIABLES_MUST_NOT_BE_NULL = "variables cannot be null";
+  static final String ERROR_EXPRESSION_EVALUATION_FAILED = "expression evaluation failed";
+
+  private final FeelEngine feelEngine;
+  private final ObjectMapper objectMapper;
+
+  public FeelEngineWrapper() {
+    this(
+        new FeelEngine.Builder()
+            .valueMapper(SpiServiceLoader.loadValueMapper())
+            .functionProvider(SpiServiceLoader.loadFunctionProvider())
+            .build(),
+        new ObjectMapper().registerModule(DefaultScalaModule$.MODULE$));
+  }
+
+  public FeelEngineWrapper(final FeelEngine feelEngine, final ObjectMapper objectMapper) {
+    this.feelEngine = feelEngine;
+    this.objectMapper = objectMapper;
+  }
+
+  private static String trimExpression(final String expression) {
+    var feelExpression = expression.trim();
+    if (feelExpression.startsWith("=")) {
+      feelExpression = feelExpression.substring(1);
+    }
+    return feelExpression.trim();
+  }
+
+  private static scala.collection.immutable.Map<String, Object> toScalaMap(
+      final Map<String, Object> responseMap) {
+    final HashMap<String, Object> context = new HashMap<>(responseMap);
+    context.put(RESPONSE_MAP_KEY, responseMap);
+    return scala.collection.immutable.Map.from(CollectionConverters.asScala(context));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static Map<String, Object> ensureVariablesMap(final Object variables) {
+    return (Map<String, Object>)
+        Objects.requireNonNull(variables, ERROR_VARIABLES_MUST_NOT_BE_NULL);
+  }
+
+  public String evaluateToJson(final String expression, final Object variables) {
+    try {
+      var variablesAsMap = ensureVariablesMap(variables);
+      var variablesAsMapAsScalaMap = toScalaMap(variablesAsMap);
+      var result = feelEngine.evalExpression(trimExpression(expression), variablesAsMapAsScalaMap);
+      if (result.isRight()) {
+        return resultToJson(result.right().get());
+      } else {
+        throw new RuntimeException(result.left().get().message());
+      }
+    } catch (Exception e) {
+      throw new FeelEngineWrapperException(
+          ERROR_EXPRESSION_EVALUATION_FAILED, expression, Objects.toString(variables), e);
```

Comment:
Do we need the `Objects#toString` here? We are passing on an object to the exception that will transformed into a String over there with `String.format(Failed to evaluate expression '%s' and context '%s', because %s", expression, context, reason)`, right?
```suggestion
          ERROR_EXPRESSION_EVALUATION_FAILED, expression, variables, e);
```

Reply:
Good point. I confused this method with more advanced equivalent from other library.